Feature/migratenetmq4 samples

drr00t opened the pull request.

drr00t committed:
update Beacon demo project to dotnet core

drr00t committed:
update HelloWorld demo to dotnet core

drr00t committed:
update Brokerless Reliability - Freelance Pattern - Model One to dotnet core

drr00t committed:
update InterBrokerRouter demo to dotnet core

drr00t committed:
update Load Balancing Pattern to dotnet core

drr00t committed:
update WeatherUpdater to dotnet core

drr00t committed:
Update the Majordomo sample implementation to dotnet core.

drr00t committed:
update tests for Majordomo dotnet core

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,18 @@
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{D0B707E4-5563-41FA-A8B0-292605332EF1}</ProjectGuid>
+    <TargetFrameworks>net45;net47;net5.0;netcoreapp3.1;netcoreapp3.1</TargetFrameworks>
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <FileVersion>1.0.0.0</FileVersion>
```
I believe this is equivalent:

```suggestion
    <Version>1.0.0.0</Assembly>
```

Also in other files.

You may like to use a `Directory.Build.props` file for:

- `Version` property
- `TargetFrameworks` property
- `PackageReference` item
 
It allows these to be specified once across the repo.

drr00t replied:
I never used for PackageReference, but if it works I agree, I will look into.

drewnoakes replied:
It will work for `PackageReference`. You can think of `Directory.Build.props` as containing regular project file elements, that is automatically inserted at the top of your project file. In that way, it can specify defaults, and a given project may override those defaults if needed.

drr00t replied:
done

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,18 @@
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{D0B707E4-5563-41FA-A8B0-292605332EF1}</ProjectGuid>
+    <TargetFrameworks>net45;net47;net5.0;netcoreapp3.1;netcoreapp3.1</TargetFrameworks>
```
`netcoreapp3.1` appears twice.

I'm curious why you specify so many target frameworks. Consider also including `net48`, as it's the highest .NET Framework version.

drr00t replied:
Hi @drewnoakes I tried to follow framework previous version, but I think that maintain net48 and net5.0 can be enough. About netcoreapp3.1, I don't know why Rider did that :)

drewnoakes replied:
Sounds good. We can revisit the set of target frameworks more easily in future if they're pulled out to a single `Directory.Build.props` file.

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,18 @@
+
   <ItemGroup>
-    <Content Include="_ReadMe.txt" />
+    <Folder Include="Properties" />
```
This `Folder` item is not needed (ditto in other projects).

drr00t replied:
done

drewnoakes commented on src/Brokerless Reliability (Freelance Pattern)/Model One/Freelance.ModelOne.Client/Properties/AssemblyInfo.cs:
```diff
@@ -1,35 +0,0 @@
-[assembly: AssemblyConfiguration("")]
-[assembly: AssemblyCompany("HP")]
-[assembly: AssemblyProduct("Freelance.ModelOne.Client")]
-[assembly: AssemblyCopyright("Copyright © HP 2013")]
```
Please review the copyright attribute here, and introduce `Copyright` properties so that copyright information is not lost as part of this change.

This is not the only file.

drr00t replied:
oh, I missed that, i will restore and include that into project file

drr00t replied:
done

drewnoakes commented on src/Majordomo/MDPClientAsyncExample/MDPClientAsyncExample.csproj:
```diff
@@ -1,80 +1,51 @@
-    <AssemblyName>MDPClientAsyncExample</AssemblyName>
+    <RootNamespace>MDPClientExample</RootNamespace>
+    <AssemblyName>MDPClientExample</AssemblyName>
     <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
```
This is at odds with the `TargetFrameworks` below. I think a lot of the contents of this file can be removed.

drr00t replied:
done

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,18 @@
+
   <ItemGroup>
-    <None Include="App.config" />
-    <None Include="packages.config" />
```
The `packages.config` files should also be deleted.

drr00t replied:
done

drewnoakes reviewed (commented):
As someone who works on the Visual Studio project system, I love to see this kind of migration. Thanks!

A few comments here for things to fix up before we can merge this.

drr00t committed:
changes from review:
- added Build props file for solution
- fixed framework supported versions
- remove duplicated netcoreapp3.1

drr00t committed:
changes:
- update Titanic Pattern examples
- removed empty properties directory from MPDClientAsyncExample

drr00t committed:
copyright:
fixup original copyright information from assemblyinfo

drr00t committed:
more copyright fixes

drr00t committed:
removing old packages.config

drr00t committed:
centralizing TargetFramework into build props file

drr00t committed:
update LazyPirate to dotnet core:
 - samples Client. both Server and Server2 versions

drr00t committed:
update ParanoidPirate to dotnet core

drr00t committed:
update SimplePirate to dotnet core

drr00t commented:
@drewnoakes i think it's for review now, but i'm didn't port two last samples Multitheading and LazyPirate.Client2 those two still broked: 

- Multitheading/Multithreaded Service: need more work because of changes in NetMQQueue implementation
- Pirate Pattern/LazyPirate.Client2: need more work because of changes in the style of pub-sub implementation.
